### PR TITLE
Create an E2E test to verify AppProject behaviour which we rely on

### PR DESCRIPTION
**Pull request by jopit**
#### Description:
- Added four new e2e tests to ensure that Argo CD behaves as expected when the repo url specified in an Application matches or doesn't match the url specified in the AppProject to which the Application belongs.

### CODE REVIEW

File: tests-e2e/argocd/argocd_appproject_test.go [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+						Namespace: fixture.GitOpsServiceE2ENamespace,
+					},
+					SyncPolicy: &appv1.SyncPolicy{
+						Automated: &appv1.SyncPolicyAutomated{},
+					},
+				},
+			}
+
+			secret = corev1.Secret{
+				TypeMeta: metav1.TypeMeta{},
```
Comment (Rizwana777) [inline tests-e2e/argocd/argocd_appproject_test.go:94]: I think this `TypeMeta` can be removed

**Comment by Rizwana777** [comment]
@jopit left a minor comment and everything LGTM, thank you

File: tests-e2e/argocd/argocd_appproject_test.go [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+					appFixture.HaveSyncStatusCode(appv1.SyncStatusCodeSynced),
+					appFixture.HaveHealthStatusCode(health.HealthStatusHealthy),
+				))
+		})
+
+		// This test requires the following environment variable to be set.
+		//		GITHUB_SSH_KEY: The SSH key of the private GitHub repository
+		// If the variable is not set, the test will be skipped.
+		It("Should succeed if the app project and the application both use a git url", func() {
+			const giturl = "git@github.com:managed-gitops-test-data/private-repo-test.git"
```
Comment (jgwest) [inline tests-e2e/argocd/argocd_appproject_test.go:134]: I'm 90% sure you already answered this in standup a little while ago, but it's always good to double check:
- I presume it's not possible to use 'git@github.com:redhat-appstudio/managed-gitops.git' here? (IIRC you said it's not)
- If my presumption is correct, could you add a quick little comment to the test that mentions that (that way the next person that comes to this test will know why we're doing it this way).

**Review by jgwest — CHANGES_REQUESTED** [review]
Looks perfect, thanks @jopit! And thanks @Rizwana777 for reviewing! One quick question (which I think has already been answered) then good to go.

File: tests-e2e/argocd/argocd_appproject_test.go (hunk 2 above)
Comment (jopit) [reply tests-e2e/argocd/argocd_appproject_test.go:137]: I've added a comment to explain.

**Review by jgwest — APPROVED** [review]
LGTM, thanks again @jopit!

**Pull request merged**